doc/permissions.md

```diff
@@ -0,0 +1,173 @@
+
+```python
+from baselayer.app.models import AccessibleIfUserMatches
+accessible_to_owner_or_last_modified_by = AccessibleIfUserMatches('owner') | AccessibleIfUserMatches('last_modified_by')
+```
+
+Policies can be custom (potentially complex and relational) database queries. This policy allows adding a user to a group only if that user has access to all of the group's streams:
+
+```
+from baselayer.app.env import CustomserAccessControl
+GroupUser.create = (
+    CustomUserAccessControl(
+        # Can only add a user to a group if they have all the requisite
+        # streams required for entry to the group
+        lambda cls, user_or_token: DBSession()
+        .query(cls)
+        .join(Group)
+        .outerjoin(Stream, Group.streams)
+        .outerjoin(
+            StreamUser,
+            sa.and_(
+                StreamUser.user_id == cls.user_id,
+                StreamUser.stream_id == Stream.id,
+            ),
+        )
+        .group_by(cls.id)
+        .having(
+            sa.or_(
+                sa.func.bool_and(StreamUser.stream_id.isnot(None)),
+                sa.func.bool_and(Stream.id.is_(None)),  # group has no streams
+            )
+        )
+    )
+)
+```
+
+## Binding Policies to Classes
+
+To bind a policy to a class, set the `create`, `read`, `update`, or `delete` attribute of the class to the policy, depending on which access mode you want the policy to apply to:
+
+```python
+Spectrum.delete = accessible_by_owner
+```
+
+The above will ensure that only the owner of a spectrum (or anyone with the "System admin" ACL) can delete the spectrum.
+
+## Convenience Methods
+
+Instances of `Base` provide convenience methods to (a) check if an instance of a mapped class is accessible to a specified user, (b) retrieve specific records of a mapped class (identified by a primary key), if they are accessible by a specified user, otherwise raise an AccessError (c) retrieve all records of a mapped class accessible to a specified user, and (d) generate a query object that returns records in a table that are accessible to a given user:
+
+(a) Can the `User` `user` read the `Spectrum` `spectrum`?
+
+```python
+spectrum.is_accessible_by(user, mode="read")
+```
+
+Can the `User` `user` update the `Spectrum` `spectrum`?
+
+```python
+spectrum.is_accessible_by(user, mode='update')
+```
+
+(b) Retrieve spectra with IDs [1, 3] if they are updatable by the `User` `user`, raising an AccessError if either does not exist:
+
+```python
+Spectrum.get_if_accessible_by([1, 3], user, raise_if_none=True, mode="update")
+```
+
+(c) Retrieve all photometry that `User` `user` can read
+
+```python
+Photometry.get_records_accessible_by(user, mode="read")
+```
+
+(d) Query for all photometry that `User` `user` can delete with a signal-to-noise ratio of at least 10:
```

Comment:
Make more explicit that a query instance is returned, not the records

Reply:
And: are these all of the base methods? If not, should show them all here

Reply:
> Make more explicit that a query instance is returned, not the records

Done

Reply:
> And: are these all of the base methods? If not, should show them all here

Yep this is everything.